master/internal/rm/agentrm/agent_state_test.go

```diff
@@ -322,6 +323,62 @@ func Test_agentState_checkAgentStartedDevicesMatch(t *testing.T) {
 	}
 }
 
+func Test_agentState_checkAgentResourcePoolMatch(t *testing.T) {
+	tests := []struct {
+		name            string
+		state           agentState
+		agentStarted    *aproto.AgentStarted
+		wantErrContains string
+	}{
+		{
+			name: "resource pool name match",
+			state: agentState{
+				resourcePoolName: "pool1",
+			},
+			agentStarted: &aproto.AgentStarted{
+				ResourcePoolName: "pool1",
```

Comment:
nit: might be better to create variables fro "pool1" and "pool2"

Reply:
Agreed. They could even be constants scoped to this test function.
```go
func Test_agentState_checkAgentResourcePoolMatch(t *testing.T) {
	const (
		poolOne = "pool1"
		poolTwo = "pool2"
	)
```